File: LeetCode/cpp/258.cpp
class Solution {
public:
    int addDigits(int num) {
        while (num > 9) {
            int sum = 0, temp = num;
            while (temp > 0) {
                sum += temp % 10;
                temp /= 10;
            }
            num = sum;
        }
        return num;
    }
};

File: LeetCode/python/258.py
class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        while num > 9:
            temp, sum = num, 0
            while temp > 0:
                sum += temp % 10
                temp //= 10
            num = sum
        return num
